File: frontend/src/services/usuarioService.js
import api from './api';


const RotaApiUsuario = '/api/usuarios';

const usuarioService = {
  listar: () => api.get(RotaApiUsuario),
  buscarPorId: (id) => api.get(`${RotaApiUsuario}/${id}`),
  criar: (usuario) => api.post(RotaApiUsuario, usuario),
  atualizar: (id, usuario) => api.put(`${RotaApiUsuario}/${id}`, usuario),
  deletar: (id) => api.delete(`${RotaApiUsuario}/${id}`),
  login: (email, senha) => api.post(`${RotaApiUsuario}/login`, { email, senha })
};

export default usuarioService;

File: fintech/src/main/java/br/com/fiap/fintech/service/UsuarioService.java
package br.com.fiap.fintech.service;

import br.com.fiap.fintech.exception.BusinessException;
import br.com.fiap.fintech.exception.ResourceNotFoundException;
import br.com.fiap.fintech.model.Usuario;
import br.com.fiap.fintech.repository.UsuarioRepository;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UsuarioService {

    private final UsuarioRepository usuarioRepository;
    private final PasswordEncoder passwordEncoder;

    public UsuarioService(UsuarioRepository usuarioRepository, PasswordEncoder passwordEncoder) {
        this.usuarioRepository = usuarioRepository;
        this.passwordEncoder = passwordEncoder;
    }

    public List<Usuario> listarTodos() {
        return usuarioRepository.findAll();
    }

    public Usuario buscarPorId(Long id) {
        return usuarioRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Usuário não encontrado com id: " + id));
    }

    public Usuario criar(Usuario usuario) {
        if (usuarioRepository.existsByEmail(usuario.getEmail())) {
            throw new BusinessException("Já existe um usuário cadastrado com este email");
        }
        usuario.setSenha(passwordEncoder.encode(usuario.getSenha()));
        return usuarioRepository.save(usuario);
    }

    public Usuario atualizar(Long id, Usuario dados) {
        Usuario existente = buscarPorId(id);
        existente.setNome(dados.getNome());
        existente.setEmail(dados.getEmail());
        if (dados.getSenha() != null && !dados.getSenha().isBlank()) {
            existente.setSenha(passwordEncoder.encode(dados.getSenha()));
        }
        return usuarioRepository.save(existente);
    }

    public void deletar(Long id) {
        usuarioRepository.delete(buscarPorId(id));
    }
}
